sale_operating_unit/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = "sale.advance.payment.inv"
+
+    def _create_invoice(self, order, so_line, amount):
+        invoice = super(SaleAdvancePaymentInv, self)._create_invoice(
+            order, so_line, amount
+        )
+        invoice.operating_unit_id = order.operating_unit_id.id
+        return invoice
```

Comment:
```suggestion
        return invoice
        
            def _prepare_invoice_values(self, order, name, amount, so_line):
        invoice_vals = super(SaleAdvancePaymentInv, self)._prepare_invoice_values(
            order, name, amount, so_line
        )
        journal = self.env["account.journal"].search(
            [
                ("operating_unit_id", "=", order.operating_unit_id.id),
                ("type", "=", "sale"),
            ],
            limit=1,
        )
        if not journal:
            raise ValidationError(
                _("You need to create a sales journal for this operating unit.")
            )
        invoice_vals.update(
            {
                "operating_unit_id": order.operating_unit_id.id,
                "journal_id": journal.id,
            }
        )
        return invoice_vals

```

I added this in order to allow to create invoices from the same OU when a user has access to multiple OU by default it shows the default OU of the user and not the OU of the sale order.